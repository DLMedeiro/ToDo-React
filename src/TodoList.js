import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";
import { v4 } from "uuid";

const TodoList = () => {
  const INITIAL_STATE = [
    { id: v4(), todo: "Learn React" },
    { id: v4(), todo: "Clean Kitchen" },
  ];

  const [todos, setTodos] = useState(INITIAL_STATE);
  const addItem = (id, todo) => {
    setTodos((todos) => [...todos, { id, todo }]);
  };
  const remove = (id) => {
    setTodos(todos.filter((t) => t.id !== id));
  };

  return (
    <div>
      <NewTodoForm addItem={addItem} />
      <div>
        {todos.map(({ id, todo }) => (
          <Todo id={id} todo={todo} remove={remove} />
        ))}
      </div>
    </div>
  );
};

export default TodoList;
