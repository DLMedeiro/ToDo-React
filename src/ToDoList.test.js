import React from "react";
import { render } from "@testing-library/react";
import TodoList from "./TodoList";

test("it renders without crashing", () => {
  render(<TodoList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

// Expect the form to add inputs to the DOM
it("works when a new toDo is added", function () {});
// Expect selected item to be removed from the DOM
it("works when a toDo is deleted", function () {});
// Expect an error to show when the form is submitted without a task entered
it("works when an incorrect input is provided", function () {});
