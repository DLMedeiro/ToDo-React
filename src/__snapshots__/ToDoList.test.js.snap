// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="todo"
      >
        New Activity
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="todo"
        type="text"
        value=""
      />
      <p />
      <button>
        Add New Item
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
